tools/pylib/boutdata/pol_slice.py

```diff
@@ -62,45 +60,26 @@ def pol_slice(var3d, gridfile, n=1, zangle=0.0, nyInterp=None):
             if zShift is not None:
                 print("Using zShift as toroidal shift angle")
             else:
-                print("ERROR: Neither qinty nor zShift found")
-                return None
-
-        gf.close()
-    except:
-        print("ERROR: pol_slice couldn't read grid file")
-        return None
-
-    #Decide if we've asked to do interpolation
-    doInterp = False
-    if nyInterp is not None:
-        if ny != nyInterp:
-            doInterp = True
-
-    #Setup interpolation if requested
-    if doInterp:
+                raise ValueError("Neither qinty nor zShift found")
 
+    # Decide if we've asked to do interpolation
+    if nyInterp is not None and nyInterp != ny:
         varTmp = var3d
-        #These are the index space co-ordinates of the output arrays
-        xOut = np.linspace(0,nx-1,nx)
-        yOut = np.linspace(0,ny-1,nyInterp)
```

Comment:
`yOut` here is a high resolution sample of y index space spanning 0 to `ny-1`, in the new version with `mgrid` seems to just use the presumably integer range `0` to `nyInterp-1`. I've not tested this but I'm not sure I'd expect this to work.

Reply:
Fixed with `0:ny-1:nyInterp*1j` which gives `nyInterp` points between 0 and `ny-1` inclusive. Confirmed that this gives identically behaviour to original implementation